### PR TITLE
GreatCircleLayer Fixes

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- Update vertex shader to match the new ArcLayer API
- Add documentation
- Tests

### CODE REVIEW

File: modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js [hunk 1]
```diff
@@ -54,11 +58,12 @@ float getSegmentRatio(float index) {
...
   vec2 delta = source - target;
+  vec2 sin_half_delta = sin(delta / 2.0);
   float a =
-    sin(delta.y / 2.0) * sin(delta.y / 2.0) +
+    sin_half_delta.y * sin_half_delta.y +
     cos(source.y) * cos(target.y) *
-    sin(delta.x / 2.0) * sin(delta.x / 2.0);
-	return 2.0 * atan(sqrt(a), sqrt(1.0 - a));
+    sin_half_delta.x * sin_half_delta.x;
+  return 2.0 * atan(sqrt(a), sqrt(1.0 - a));
```
Comment (xintongxia) [inline modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js]: format

File: modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js [hunk 2]
```diff
@@ -69,9 +74,14 @@ vec2 interpolate (vec2 source, vec2 target, float angularDist, float t) {
...
-	float y = a * cos(source.y) * sin(source.x) + b * cos(target.y) * sin(target.x);
-	float z = a * sin(source.y) + b * sin(target.y);
+  vec2 sin_source = sin(source);
+  vec2 cos_source = cos(source);
+  vec2 sin_target = sin(target);
+  vec2 cos_target = cos(target);
+
+  float x = a * cos_source.y * cos_source.x + b * cos_target.y * cos_target.x;
+  float y = a * cos_source.y * sin_source.x + b * cos_target.y * sin_target.x;
+  float z = a * sin_source.y + b * sin_target.y;
```
Comment (xintongxia) [inline modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js]: Line 77 - 84 indent not correct

File: modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js [hunk 3]
```diff
@@ -54,11 +58,12 @@ float getSegmentRatio(float index) {
...
-    sin(delta.y / 2.0) * sin(delta.y / 2.0) +
+    sin_half_delta.y * sin_half_delta.y +
     cos(source.y) * cos(target.y) *
-    sin(delta.x / 2.0) * sin(delta.x / 2.0);
-	return 2.0 * atan(sqrt(a), sqrt(1.0 - a));
+    sin_half_delta.x * sin_half_delta.x;
+  return 2.0 * atan(sqrt(a), sqrt(1.0 - a));
 }
 
 vec2 interpolate (vec2 source, vec2 target, float angularDist, float t) {
```
Comment (xintongxia) [inline modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js]: Maybe link to a page to explain about the spherical interpolation.

File: modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js (hunk 2 above)
Comment (Pessimistress) [reply modules/geo-layers/src/great-circle-layer/great-circle-vertex.glsl.js]: It's just github formatting tabs weirdly

**Pull request merged**